### ISSUE TITLE
Inceptionv1 doesn't converge when training on a 24-node cluster with bs 3072

### ISSUE THREAD

**Issue by nwang2** [intel-analytics/BigDL#2494]
--executor-cores=32 
--total-executor-cores=768
--batchSize=3072
We see loss is NaN in the log:
2018-04-14 00:58:52 INFO  DistriOptimizer$:392 - [Epoch 149 390144/1281167][Iteration 61991][Wall Clock 90297.102702327s] Trained 3072 records in 1.402170578 seconds. Throughput is 2190.889 records/second. Loss is NaN. Current learning rate is 0.0011379211379215811. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-14 00:58:53 INFO  DistriOptimizer$:392 - [Epoch 149 393216/1281167][Iteration 61992][Wall Clock 90298.51743813s] Trained 3072 records in 1.414735803 seconds. Throughput is 2171.4302 records/second. Loss is NaN. Current learning rate is 0.0010795267780450456. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-14 00:58:55 INFO  DistriOptimizer$:392 - [Epoch 149 396288/1281167][Iteration 61993][Wall Clock 90299.903990064s] Trained 3072 records in 1.386551934 seconds. Throughput is 2215.5679 records/second. Loss is NaN. Current learning rate is 0.0010177876069706277. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-14 00:58:56 INFO  DistriOptimizer$:392 - [Epoch 149 399360/1281167][Iteration 61994][Wall Clock 90301.292274874s] Trained 3072 records in 1.38828481 seconds. Throughput is 2212.8025 records/second. Loss is NaN. Current learning rate is 9.520531294473895E-4. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-14 00:58:57 INFO  DistriOptimizer$:392 - [Epoch 149 402432/1281167][Iteration 61995][Wall Clock 90302.739086831s] Trained 3072 records in 1.446811957 seconds. Throughput is 2123.289 records/second. Loss is NaN. Current learning rate is 8.814299232935354E-4. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-14 00:58:59 INFO  DistriOptimizer$:392 - [Epoch 149 405504/1281167][Iteration 61996][Wall Clock 90304.218591304s] Trained 3072 records in 1.479504473 seconds. Throughput is 2076.3708 records/second. Loss is NaN. Current learning rate is 8.046317530801396E-4. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-14 00:59:00 INFO  DistriOptimizer$:392 - [Epoch 149 408576/1281167][Iteration 61997][Wall Clock 90305.630878178s] Trained 3072 records in 1.412286874 seconds. Throughput is 2175.1956 records/second. Loss is NaN. Current learning rate is 7.196845186965593E-4. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-14 00:59:02 INFO  DistriOptimizer$:392 - [Epoch 149 411648/1281167][Iteration 61998][Wall Clock 90307.028659539s] Trained 3072 records in 1.397781361 seconds. Throughput is 2197.7686 records/second. Loss is NaN. Current learning rate is 6.232650759019548E-4. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-14 00:59:03 INFO  DistriOptimizer$:392 - [Epoch 149 414720/1281167][Iteration 61999][Wall Clock 90308.492408698s] Trained 3072 records in 1.463749159 seconds. Throughput is 2098.7202 records/second. Loss is NaN. Current learning rate is 5.088938034853138E-4. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-14 00:59:14 INFO  DistriOptimizer$:392 - [Epoch 149 417792/1281167][Iteration 62000][Wall Clock 90310.11265655s] Trained 3072 records in 1.620247852 seconds. Throughput is 1896.0062 records/second. Loss is NaN. Current learning rate is 3.598422593488989E-4. Current weight decay is 1.0E-4. Current momentum is 0.9.

**Comment by qiuxin2012** [comment]
Could you also attach the logs before `NaN`? @nwang2

**Comment by nwang2** [comment]
2018-04-13 07:12:15 INFO  DistriOptimizer$:392 - [Epoch 42 933888/1281167][Iteration 17442][Wall Clock 26290.6678436s] Trained 3072 records in 1.468526684 seconds. Throughput is 2091.8926 records/second. Loss is 1.72319. Current learning rate is 0.07595911266872037. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:17 INFO  DistriOptimizer$:392 - [Epoch 42 936960/1281167][Iteration 17443][Wall Clock 26292.140475331s] Trained 3072 records in 1.472631731 seconds. Throughput is 2086.0613 records/second. Loss is 1.9208802. Current learning rate is 0.07595826032083502. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:18 INFO  DistriOptimizer$:392 - [Epoch 42 940032/1281167][Iteration 17444][Wall Clock 26293.563798395s] Trained 3072 records in 1.423323064 seconds. Throughput is 2158.3293 records/second. Loss is 1.98872. Current learning rate is 0.07595740796338517. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:20 INFO  DistriOptimizer$:392 - [Epoch 42 943104/1281167][Iteration 17445][Wall Clock 26295.038505653s] Trained 3072 records in 1.474707258 seconds. Throughput is 2083.1252 records/second. Loss is 2.7426317. Current learning rate is 0.07595655559637046. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:21 INFO  DistriOptimizer$:392 - [Epoch 42 946176/1281167][Iteration 17446][Wall Clock 26296.456909455s] Trained 3072 records in 1.418403802 seconds. Throughput is 2165.815 records/second. Loss is 3.159315. Current learning rate is 0.07595570321979057. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:23 INFO  DistriOptimizer$:392 - [Epoch 42 949248/1281167][Iteration 17447][Wall Clock 26297.94669795s] Trained 3072 records in 1.489788495 seconds. Throughput is 2062.0376 records/second. Loss is 4.7848105. Current learning rate is 0.07595485083364517. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:24 INFO  DistriOptimizer$:392 - [Epoch 42 952320/1281167][Iteration 17448][Wall Clock 26299.439473033s] Trained 3072 records in 1.492775083 seconds. Throughput is 2057.912 records/second. Loss is 4.623628. Current learning rate is 0.07595399843793399. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:25 INFO  DistriOptimizer$:392 - [Epoch 42 955392/1281167][Iteration 17449][Wall Clock 26300.892232065s] Trained 3072 records in 1.452759032 seconds. Throughput is 2114.5972 records/second. Loss is 5.4375186. Current learning rate is 0.07595314603265664. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:27 INFO  DistriOptimizer$:392 - [Epoch 42 958464/1281167][Iteration 17450][Wall Clock 26302.415386717s] Trained 3072 records in 1.523154652 seconds. Throughput is 2016.8668 records/second. Loss is 7.1270523. Current learning rate is 0.07595229361781282. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:28 INFO  DistriOptimizer$:392 - [Epoch 42 961536/1281167][Iteration 17451][Wall Clock 26303.812892377s] Trained 3072 records in 1.39750566 seconds. Throughput is 2198.2021 records/second. Loss is 6.8692684. Current learning rate is 0.07595144119340223. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:30 INFO  DistriOptimizer$:392 - [Epoch 42 964608/1281167][Iteration 17452][Wall Clock 26305.268950912s] Trained 3072 records in 1.456058535 seconds. Throughput is 2109.8054 records/second. Loss is 6.897251. Current learning rate is 0.07595058875942454. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:31 INFO  DistriOptimizer$:392 - [Epoch 42 967680/1281167][Iteration 17453][Wall Clock 26306.588188593s] Trained 3072 records in 1.319237681 seconds. Throughput is 2328.6174 records/second. Loss is 6.902765. Current learning rate is 0.07594973631587942. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:33 INFO  DistriOptimizer$:392 - [Epoch 42 970752/1281167][Iteration 17454][Wall Clock 26308.020594884s] Trained 3072 records in 1.432406291 seconds. Throughput is 2144.6428 records/second. Loss is 6.8997383. Current learning rate is 0.07594888386276655. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:34 INFO  DistriOptimizer$:392 - [Epoch 42 973824/1281167][Iteration 17455][Wall Clock 26309.392353916s] Trained 3072 records in 1.371759032 seconds. Throughput is 2239.4604 records/second. Loss is 7.1552887. Current learning rate is 0.07594803140008559. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:35 INFO  DistriOptimizer$:392 - [Epoch 42 976896/1281167][Iteration 17456][Wall Clock 26310.786160545s] Trained 3072 records in 1.393806629 seconds. Throughput is 2204.0361 records/second. Loss is 6.930203. Current learning rate is 0.07594717892783626. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:37 INFO  DistriOptimizer$:392 - [Epoch 42 979968/1281167][Iteration 17457][Wall Clock 26312.251475085s] Trained 3072 records in 1.46531454 seconds. Throughput is 2096.4783 records/second. Loss is 6.907976. Current learning rate is 0.07594632644601818. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:39 INFO  DistriOptimizer$:392 - [Epoch 42 983040/1281167][Iteration 17458][Wall Clock 26314.280752352s] Trained 3072 records in 2.029277267 seconds. Throughput is 1513.8394 records/second. Loss is 6.9103928. Current learning rate is 0.07594547395463108. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:40 INFO  DistriOptimizer$:392 - [Epoch 42 986112/1281167][Iteration 17459][Wall Clock 26315.754152692s] Trained 3072 records in 1.47340034 seconds. Throughput is 2084.9731 records/second. Loss is 6.9094315. Current learning rate is 0.07594462145367462. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:42 INFO  DistriOptimizer$:392 - [Epoch 42 989184/1281167][Iteration 17460][Wall Clock 26317.311533589s] Trained 3072 records in 1.557380897 seconds. Throughput is 1972.5425 records/second. Loss is 6.9113755. Current learning rate is 0.07594376894314847. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:43 INFO  DistriOptimizer$:392 - [Epoch 42 992256/1281167][Iteration 17461][Wall Clock 26318.705029521s] Trained 3072 records in 1.393495932 seconds. Throughput is 2204.5273 records/second. Loss is 6.9081864. Current learning rate is 0.07594291642305231. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:45 INFO  DistriOptimizer$:392 - [Epoch 42 995328/1281167][Iteration 17462][Wall Clock 26320.103285948s] Trained 3072 records in 1.398256427 seconds. Throughput is 2197.022 records/second. Loss is 6.910982. Current learning rate is 0.07594206389338584. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:46 INFO  DistriOptimizer$:392 - [Epoch 42 998400/1281167][Iteration 17463][Wall Clock 26321.566087733s] Trained 3072 records in 1.462801785 seconds. Throughput is 2100.0793 records/second. Loss is 6.9067855. Current learning rate is 0.0759412113541487. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:48 INFO  DistriOptimizer$:392 - [Epoch 42 1001472/1281167][Iteration 17464][Wall Clock 26322.968088617s] Trained 3072 records in 1.402000884 seconds. Throughput is 2191.154 records/second. Loss is 6.952673. Current learning rate is 0.0759403588053406. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:49 INFO  DistriOptimizer$:392 - [Epoch 42 1004544/1281167][Iteration 17465][Wall Clock 26324.421988442s] Trained 3072 records in 1.453899825 seconds. Throughput is 2112.9377 records/second. Loss is 6.943621. Current learning rate is 0.07593950624696119. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:51 INFO  DistriOptimizer$:392 - [Epoch 42 1007616/1281167][Iteration 17466][Wall Clock 26326.005688678s] Trained 3072 records in 1.583700236 seconds. Throughput is 1939.7611 records/second. Loss is 7.1277766. Current learning rate is 0.07593865367901016. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:52 INFO  DistriOptimizer$:392 - [Epoch 42 1010688/1281167][Iteration 17467][Wall Clock 26327.454681169s] Trained 3072 records in 1.448992491 seconds. Throughput is 2120.0938 records/second. Loss is 21.49688. Current learning rate is 0.0759378011014872. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:53 INFO  DistriOptimizer$:392 - [Epoch 42 1013760/1281167][Iteration 17468][Wall Clock 26328.811476349s] Trained 3072 records in 1.35679518 seconds. Throughput is 2264.159 records/second. Loss is 96.20657. Current learning rate is 0.07593694851439199. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:55 INFO  DistriOptimizer$:392 - [Epoch 42 1016832/1281167][Iteration 17469][Wall Clock 26330.124177676s] Trained 3072 records in 1.312701327 seconds. Throughput is 2340.2124 records/second. Loss is 1.52064655E10. Current learning rate is 0.07593609591772418. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:56 INFO  DistriOptimizer$:392 - [Epoch 42 1019904/1281167][Iteration 17470][Wall Clock 26331.5463243s] Trained 3072 records in 1.422146624 seconds. Throughput is 2160.1147 records/second. Loss is NaN. Current learning rate is 0.07593524331148346. Current weight decay is 1.0E-4. Current momentum is 0.9.
2018-04-13 07:12:58 INFO  DistriOptimizer$:392 - [Epoch 42 1022976/1281167][Iteration 17471][Wall Clock 26333.00896332s] Trained 3072 records in 1.46263902 seconds. Throughput is 2100.3132 records/second. Loss is NaN. Current learning rate is 0.07593439069566951. Current weight decay is 1.0E-4. Current momentum is 0.9.

**Comment by qiuxin2012** [comment]
@nwang2 Seems a random failure, could you resume the training at epoch 41?
